boards: add support for makediary m2 devkit

purpl3F0x opened the pull request:
This PR adds support for makerdiary M.2 Developer Kit.



PS: Arguably the SPI3 is a better choice for the LCD and SD card. But I left it as it was on the manufacturers repo. It should probably be changed.

purpl3F0x committed:
boards: add support for makediary m2 devkit

Adds support for the Makerdiary nRF52840 M2 devkit.

Signed-off-by: Stavros Avramidis <stavros9899@gmail.com>

nordicjm commented on boards/makerdiary/nrf52832_mdk/nrf52840_m2_defconfig:
```diff
@@ -0,0 +1,31 @@
+# Copyright (c) 2022 Stavros Avramidis <stavros9899@gmail.com>
+# SPDX-License-Identifier: Apache-2.0
+
+CONFIG_SOC_SERIES_NRF52X=y
+CONFIG_SOC_NRF52840_QIAA=y
+CONFIG_BOARD_NRF52840_M2=y
```
this is hwmv1, this board needs adding in hwmv2

purpl3F0x replied:
I also placed it in the folder of another board 🥲😅

purpl3F0x replied:
I amended the commit to be compliant with hmmv2

nordicjm commented on boards/makerdiary/nrf52840_m2/doc/img/nrf52840-m2-devkit-diagram.jpg:
Use webp for images, and put them through https://tinypng.com/

purpl3F0x replied:
The diagram picture can also be in SVG format, is this preferred ?

Is this the correct link I should be looking at, for image guidelines ? so smaller than 1000pxwidth  and 250KB size https://docs.zephyrproject.org/latest/contribute/documentation/guidelines.html

nordicjm replied:
@kartben

nordicjm reviewed (commented):
A commit message is needed

nordicjm commented on boards/makerdiary/nrf52840_m2/Kconfig.defconfig:
```diff
@@ -0,0 +1,11 @@
+config BT_CTLR
+	default BT
+
+endif # BOARD_NRF52840_M2
```
missing newline at end of file

nordicjm commented on boards/makerdiary/nrf52840_m2/Kconfig.nrf52840_m2:
```diff
@@ -0,0 +1,6 @@
+# Copyright (c) 2024 Stavros Avramidis <stavros9899@gmail.com>
+# SPDX-License-Identifier: Apache-2.0
+
+config BOARD_NRF52840_M2
+	bool "nrf52840_m2"
+	depends on SOC_NRF52840_QIAA
```
this is not hwmv2

nordicjm commented on boards/makerdiary/nrf52840_m2/board.cmake:
```diff
@@ -0,0 +1,8 @@
+set(OPENOCD_NRF5_INTERFACE "cmsis-dap")
+
+include(${ZEPHYR_BASE}/boards/common/pyocd.board.cmake)
+include(${ZEPHYR_BASE}/boards/common/openocd-nrf5.board.cmake)
```
many files missing newlines at end of them

nordicjm commented on boards/makerdiary/nrf52840_m2/doc/img/nrf52840-m2-devkit-diagram.jpg:
convert images to webp and put through https://tinypng.com/

purpl3F0x replied:
Would it be better for the diagram to be put in .svg for optimal file size and resolution ?

nordicjm commented on boards/makerdiary/nrf52840_m2/doc/index.rst:
```diff
@@ -0,0 +1,90 @@
+
+.. |image2| image:: img/nrf52840-m2-devkit-diagram.jpg
+   :target: https://wiki.makerdiary.com/nrf52840-m2-devkit/resources/nrf52840_m2_devkit_hw_diagram_v1_0.pdf
+
```
might be a good idea to mention how to program it, does `west debug` work or do you need to copy a .uf2 file to a drive or what

purpl3F0x replied:
It can be done both ways, the makerdiary MDK docs doesn't mention something, and it's more of the same board programming wise.

nordicjm commented on boards/makerdiary/nrf52840_m2/nrf52840_m2.dts:
```diff
@@ -0,0 +1,317 @@
+};
+
+&uart0 {
+	compatible = "nordic,nrf-uart";
```
```suggestion
	compatible = "nordic,nrf-uarte";
```

nordicjm commented on boards/makerdiary/nrf52840_m2/nrf52840_m2.dts:
```diff
@@ -0,0 +1,317 @@
+	pinctrl-names = "default", "sleep";
+};
+
+arduino_i2c: &i2c0 {
+	compatible = "nordic,nrf-twi";
+	status = "okay";
+	pinctrl-0 = <&i2c0_default>;
+	pinctrl-1 = <&i2c0_sleep>;
+	pinctrl-names = "default", "sleep";
+};
+
+&i2c1 {
+	compatible = "nordic,nrf-twi";
+	status = "okay";
+	pinctrl-0 = <&i2c1_default>;
+	pinctrl-1 = <&i2c1_sleep>;
+	pinctrl-names = "default", "sleep";
+};
+
+&pwm0 {
+	status = "okay";
+	pinctrl-0 = <&pwm0_default>;
+	pinctrl-1 = <&pwm0_sleep>;
+	pinctrl-names = "default", "sleep";
+};
+
+arduino_spi: &spi1 {
+	compatible = "nordic,nrf-spi";
+	status = "okay";
+	pinctrl-0 = <&spi1_default>;
+	pinctrl-1 = <&spi1_sleep>;
+	pinctrl-names = "default", "sleep";
+	cs-gpios = <&arduino_header 16 0>; /* D10 */
+};
+
+
+&spi0 {
```
these conflict, you can have either i2c0 or spi0 but not both, likewise i2c1 or spi1 but not both

purpl3F0x replied:
So, I could move spi0 -> spi3 and spi1 to spi 2 ?

Also, they have put the LED0 on the same pin as SCK, idk what's up with that.

nordicjm replied:
Hmm I don't see that, led0 is on p1.07 and the only other pin with that is d13 on the arduino connector: https://cdn.shopifycdn.net/s/files/1/0066/0865/0355/files/nrf52840-m2-devkit_diagram.png?v=1591971331

nordicjm replied:
Yes, spi3 and spi2 can be used (or both can use spi3) and i2c0 and i2c1 can be enabled

purpl3F0x replied:
> Hmm I don't see that, led0 is on p1.07 and the only other pin with that is d13 on the arduino connector: https://cdn.shopifycdn.net/s/files/1/0066/0865/0355/files/nrf52840-m2-devkit_diagram.png?v=1591971331

D13 if not mistaken, for Arduino hats, should be SCK.

nordicjm commented on boards/makerdiary/nrf52840_m2/nrf52840_m2.dts:
```diff
@@ -0,0 +1,317 @@
+	pinctrl-names = "default", "sleep";
+	cs-gpios = <&gpio0 6 0>,<&gpio0 7 0>;
+
+	st7789v@0 {
```
might want to move these to spi3 and use `	compatible = "nordic,nrf-spim";` as spim3 is the fast interface that supports 32MHz instead of 8MHz

purpl3F0x replied:
Yes, I mentioned it in the PR description, that it could be changed. Manufacturer repo had it as is. I will change it.

nordicjm commented on boards/makerdiary/nrf52840_m2/nrf52840_m2.dts:
```diff
@@ -0,0 +1,317 @@
+		};
+	};
+};
+
```
```suggestion
```

nordicjm commented on boards/makerdiary/nrf52840_m2/nrf52840_m2.dts:
```diff
@@ -0,0 +1,317 @@
+		t-exit-dpd = <35000>;
+	};
+};
+
```
```suggestion
```

nordicjm commented on boards/makerdiary/nrf52840_m2/nrf52840_m2.dts:
```diff
@@ -0,0 +1,317 @@
+			label = "mcuboot";
+			reg = <0x00000000 0x0000C000>;
+		};
+		slot0_partition: partition@c000 {
+			label = "image-0";
+			reg = <0x0000C000 0x00067000>;
+		};
+		slot1_partition: partition@73000 {
+			label = "image-1";
+			reg = <0x00073000 0x00067000>;
+		};
```
what mode are you using mcuboot in? If you are going to use the default, swap using move, then these sectors are not optimal. It is up to you but I would suggest moving slot1 to the qspi flash, that way you can effectively double the size of an image. Please build mcuboot with debug logging enabled and check the output at boot up, it will emit a debug log about the number of sectors

purpl3F0x replied:
I will make it the same as
https://github.com/zephyrproject-rtos/zephyr/blob/main/boards/makerdiary/nrf52840_mdk/nrf52840_mdk.dts

nordicjm commented on boards/makerdiary/nrf52840_m2/nrf52840_m2.yaml:
```diff
@@ -0,0 +1,25 @@
+identifier: nrf52840_m2
```
@anangl think this is too common of a name and makerdiary should be in it or ok as-is?

nordicjm commented on boards/makerdiary/nrf52840_m2/nrf52840_m2_defconfig:
```diff
@@ -0,0 +1,27 @@
+# Enable RTT
+CONFIG_USE_SEGGER_RTT=y
+
+# enable GPIO
```
```suggestion
# Enable GPIO
```

nordicjm commented on boards/makerdiary/nrf52840_m2/nrf52840_m2_defconfig:
```diff
@@ -0,0 +1,27 @@
+# Enable UART driver
+CONFIG_SERIAL=y
+
+# enable console
```
```suggestion
# Enable console
```

nordicjm commented on boards/makerdiary/nrf52840_m2/nrf52840_m2_defconfig:
```diff
@@ -0,0 +1,27 @@
+CONFIG_CONSOLE=y
+CONFIG_UART_CONSOLE=y
+
+# additional board options
```
```suggestion
# Additional board options
```

purpl3F0x commented:
I amended the commit with the following changes:
- LCD + SDcard are now on SPI3M
- SPI1 -> SPI2, to avoid conflicts with I2C
- Added new line to all the source files
- Replace the images with .webp
- Change boot flash config, as per nrf52840_mdk (uses the same SoC and flash).
- Changed to `uarte`

Also:
- Maybe docs should include/have a link, to the power profiling instructions.
